src/main/java/org/primeframework/mvc/security/Encryptor.java

```diff
@@ -23,17 +23,30 @@
 public interface Encryptor {
   /**
    * Decrypt a set of bytes.
+   * <p>
+   * This will attempt to decrypt with GCM and then CBC if that fails.
    *
    * @param bytes The bytes to decrypt.
    * @return The decrypted bytes.
    */
   byte[] decrypt(byte[] bytes) throws Exception;
 
   /**
-   * Encrypt a set of bytes.
+   * Encrypt a set of bytes using AES/CBC.
+   * <p>
+   * This method is provided for backward compatibility. Prefer using the more secure {@link #encryptGCM(byte[])} unless CBC mode is required.
    *
    * @param bytes The bytes to encrypt.
    * @return The encrypted bytes.
    */
+  @Deprecated(since = "4.24.0")
   byte[] encrypt(byte[] bytes) throws Exception;
+
+  /**
+   * Encrypt a set of bytes using AES/GCM.
+   *
+   * @param bytes The bytes to encrypt.
+   * @return The encrypted bytes.
+   */
+  byte[] encryptGCM(byte[] bytes) throws Exception;
```

Comment:
Can we delete this and just keep the `encrypt` method? All of the business logic about how to encrypt and decrypt stuff should be inside of `DefaultEncryptor`, so ideally the caller would not need to know anything changed. 

All existing code that calls `encrypt` should continue to work the same, it will just now use GCM, and all callers of `decrypt` should continue to work the same, only the impl of `decrypt` will try GCM and fall back. 

Perhaps there is an edge case I am not accounting for?